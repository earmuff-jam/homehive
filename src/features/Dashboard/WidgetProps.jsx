import { Box } from "@mui/material";
import InvoiceTimelineChart from "src/features/Widgets/InvoiceTimelineChart";
import InvoiceTrendsChart from "src/features/Widgets/InvoiceTrends";
import ItemTypeFreqChart from "src/features/Widgets/ItemTypeFreqChart";

export default function WidgetProps(widget) {
  switch (widget.id) {
    case 1:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTimelineChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 2:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTrendsChart
            label={widget.label}
            caption={widget.caption}
          />
        </Box>
      );
    case 3:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <ItemTypeFreqChart
            label={widget.label}
            caption={widget.caption}
          />
        </Box>
      );
    default:
      return null;
  }
}
