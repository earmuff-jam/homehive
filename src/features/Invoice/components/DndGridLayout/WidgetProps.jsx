import React from "react";

import { Box } from "@mui/material";
import DetailsTableView from "features/Invoice/components/Widgets/DetailsTableView";
import InvoiceTimelineChart from "features/Invoice/components/Widgets/InvoiceTimelineChart";
import InvoiceTrendsChart from "features/Invoice/components/Widgets/InvoiceTrends";
import ItemTypeFreqChart from "features/Invoice/components/Widgets/ItemTypeFreqChart";

export default function WidgetProps(widget) {
  switch (widget.id) {
    case 1:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTimelineChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 2:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTrendsChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 3:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <ItemTypeFreqChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 4:
      return <DetailsTableView label={widget.label} caption={widget.caption} />;
    default:
      return null;
  }
}
