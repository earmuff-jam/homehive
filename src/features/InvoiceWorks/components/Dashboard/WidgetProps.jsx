import { Box } from "@mui/material";
import DetailsTableView from "features/InvoiceWorks/components/Widgets/DetailsTableView";
import InvoiceTimelineChart from "features/InvoiceWorks/components/Widgets/InvoiceTimelineChart";
import InvoiceTrendsChart from "features/InvoiceWorks/components/Widgets/InvoiceTrends";
import ItemTypeFreqChart from "features/InvoiceWorks/components/Widgets/ItemTypeFreqChart";

export default function WidgetProps(widget) {
  switch (widget.id) {
    case 1:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTimelineChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 2:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <InvoiceTrendsChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 3:
      return (
        <Box sx={{ height: "20rem", width: "40rem" }}>
          <ItemTypeFreqChart label={widget.label} caption={widget.caption} />
        </Box>
      );
    case 4:
      return <DetailsTableView label={widget.label} caption={widget.caption} />;
    default:
      return null;
  }
}
