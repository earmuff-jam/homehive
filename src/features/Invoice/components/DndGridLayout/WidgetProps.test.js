import React from "react";
import { render } from "@testing-library/react";
import WidgetProps from "./WidgetProps";

jest.mock("features/Invoice/components/Widgets/DetailsTableView", () => () => (
  <div>Mocked DetailsTableView</div>
));
jest.mock("features/Invoice/components/Widgets/InvoiceTimelineChart", () => () => (
  <div>Mocked InvoiceTimelineChart</div>
));
jest.mock("features/Invoice/components/Widgets/InvoiceTrends", () => () => (
  <div>Mocked InvoiceTrendsChart</div>
));
jest.mock("features/Invoice/components/Widgets/ItemTypeFreqChart", () => () => (
  <div>Mocked ItemTypeFreqChart</div>
));

describe("WidgetProps", () => {
  const baseWidget = { label: "Label", caption: "Caption" };

  it("renders InvoiceTimelineChart when id=1", () => {
    const { asFragment, getByText } = render(WidgetProps({ ...baseWidget, id: 1 }));
    expect(getByText(/Mocked InvoiceTimelineChart/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders InvoiceTrendsChart when id=2", () => {
    const { asFragment, getByText } = render(WidgetProps({ ...baseWidget, id: 2 }));
    expect(getByText(/Mocked InvoiceTrendsChart/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders ItemTypeFreqChart when id=3", () => {
    const { asFragment, getByText } = render(WidgetProps({ ...baseWidget, id: 3 }));
    expect(getByText(/Mocked ItemTypeFreqChart/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders DetailsTableView when id=4", () => {
    const { asFragment, getByText } = render(WidgetProps({ ...baseWidget, id: 4 }));
    expect(getByText(/Mocked DetailsTableView/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders null for unknown id", () => {
    const { asFragment, container } = render(WidgetProps({ ...baseWidget, id: 999 }));
    expect(container.firstChild).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});
