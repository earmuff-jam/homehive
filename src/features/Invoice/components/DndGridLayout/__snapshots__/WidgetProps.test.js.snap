// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetProps renders DetailsTableView when id=4 1`] = `
<DocumentFragment>
  <div>
    Mocked DetailsTableView
  </div>
</DocumentFragment>
`;

exports[`WidgetProps renders InvoiceTimelineChart when id=1 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16x70mu"
  >
    <div>
      Mocked InvoiceTimelineChart
    </div>
  </div>
</DocumentFragment>
`;

exports[`WidgetProps renders InvoiceTrendsChart when id=2 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16x70mu"
  >
    <div>
      Mocked InvoiceTrendsChart
    </div>
  </div>
</DocumentFragment>
`;

exports[`WidgetProps renders ItemTypeFreqChart when id=3 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16x70mu"
  >
    <div>
      Mocked ItemTypeFreqChart
    </div>
  </div>
</DocumentFragment>
`;

exports[`WidgetProps renders null for unknown id 1`] = `<DocumentFragment />`;
